reactive/context_test.ts

```diff
@@ -183,3 +215,486 @@ test("all")("Scope.target works as expected when run within a `with` context", (
   expect(listeners.call).toHaveBeenCalledTimes(1)
   expect(observable.foo).toBe(true)
 })
+
+test("all")("Scope.target works with isolated and shared properties", () => {
+  const { context, listeners } = observe({
+    setOfUrls: new Set<string>(["https://example.com"]),
+    name: "ParentContext",
+  })
+
+  const childContext = context.with({
+    name: "ChildContext",
+    isolatedSetOfUrls: new Set<string>(),
+  })
+
+  // Access the isolated set in the child context
+  childContext.target.isolatedSetOfUrls.add("isolated-url")
+  expect(childContext.target.isolatedSetOfUrls.size).toBe(1)
+  expect(childContext.target.isolatedSetOfUrls.has("isolated-url")).toBe(true)
+
+  // Access the shared set in both contexts
+  childContext.target.setOfUrls.add("https://child.com")
+  expect(context.target.setOfUrls.has("https://child.com")).toBe(true)
+  expect(listeners.set).toHaveBeenCalledTimes(0) // No shared change event is fired
+})
+
+test("all")("Scope.target handles deep inheritance of properties", () => {
+  const { context } = observe({ foo: "parent value" })
+  const childContext = context.with({ bar: "child value" })
+  const grandchildContext = childContext.with({ baz: "grandchild value" })
+
+  expect(grandchildContext.target.foo).toBe("parent value")
+  expect(grandchildContext.target.bar).toBe("child value")
+  expect(grandchildContext.target.baz).toBe("grandchild value")
+
+  grandchildContext.target.foo = "updated value"
+  expect(context.target.foo).toBe("updated value")
+  expect(childContext.target.foo).toBe("updated value")
+  expect(grandchildContext.target.foo).toBe("updated value")
```

Comment:
```suggestion
  expect(grandchildContext.target.foo).toBe("updated value")
  
  context.target.foo = "updated value 2"
  expect(context.target.foo).toBe("updated value 2")
  expect(childContext.target.foo).toBe("updated value 2")
  expect(grandchildContext.target.foo).toBe("updated value 2")
```

Possible to add these assertions to test the bi-directionality (from ancestors to descendants) too ?

Reply:
Done